File: Python_devo/golden_birthday_year.py
#!/usr/local/bin/python3

import argparse

class GoldenBirthdayYear:

    def __init__(self, birth_year):
        self.birth_year = birth_year

    def calculate_golden_birthday_year(self):
        birth_digits = self.birth_year % 100
        golden_year_int = self.birth_year + birth_digits
        return golden_year_int

    def print_golden_birthday_year(self, pprint):
        gby = self.calculate_golden_birthday_year()
        if pprint:
            output = "Since your birth year is " + str(self.birth_year) + ", your golden birthday is " + str(gby)
        else:
            output = gby
        print(output)
        return output

if __name__ == '__main__':
    args_parser = argparse.ArgumentParser(description='Calculates the golden birth, which is the birth year added to the birth digits, so someone born in 2004 would have their golden birthday in 2004 + 4 = 2008.',epilog='Happy birthday, whenever! You have only one of these.')
    args_parser.add_argument('-p', '--pprint', action='store_true', default=False)
    args_stuff = args_parser.parse_known_args()
    args = args_stuff[0]
    birthdays = args_stuff[1]
    
    for birthday in birthdays:
        gby = GoldenBirthdayYear(int(birthday))
        gby.print_golden_birthday_year(args.pprint)

File: src/main/java/org/visualphysics/GoldenBirthdayYear.java
/* Copyright 2012 by Douglas Sweetser, sweetser@alum.mit.edu
 * Licensed under the Apache License, Version 2.0.
 */

package org.visualphysics;

import com.beust.jcommander.JCommander;
import java.util.ArrayList;
import java.util.List;

/** Description for Javadoc
 *  @author doug  sweetser@alum.mit.edu
 */
public class GoldenBirthdayYear {

    String birthyear;

    public GoldenBirthdayYear(String by) {
        birthyear = by;
    }

    public String calculate_golden_birthday_year() {
        String birth_digits = this.birthyear.substring(2);
        int golden_year_int = Integer.valueOf(birth_digits) + Integer.valueOf(this.birthyear);
        return Integer.toString(golden_year_int);
    }

    public String print_golden_birthday_year(boolean pprint){
        String gby = this.calculate_golden_birthday_year();
        String output;
        if(pprint) {
            output = "Since your birth year is " + this.birthyear + ", your golden birthday is " + gby;
        }
        else {
            output = gby;
        }
        System.out.println(output);
        return output;
    }

    public static void main(String[] args) {
        GoldenBirthdayYearArgs params = new GoldenBirthdayYearArgs();
        JCommander cmd = new JCommander(params, args);

        if(params.help) {
            System.out.println("Calculates the golden birth, which is the birth year added to the birth digits, so someone born in 2004 would have their golden birthday in 2004 + 4 = 2008.\n -p --pprint   pretty print\n -h --help     print help \nHappy birthday, whenever! You have only one of these.");
        }
        for (int i = 0; i < params.argv.size(); i++) {
            GoldenBirthdayYear gby = new GoldenBirthdayYear(params.argv.get(i));
            gby.print_golden_birthday_year(params.pprint);
        }
    }
}


